array_methods_lab.js

```diff
@@ -0,0 +1,308 @@
+// # Exercises
+//
+// 1. Given an array of numbers, write a function that returns a new array where every element is incremented by 10.
+//
+// ```
+// Input: [1,2,3,4,5]
+// Output: [11,12,13,14,15]
+// ```
+// USE MAP METHOD
+let arr1 = [2, 3, 4, 11, 5]
+let increasedBy10 = arr1.map((currentElem)=>{
+  return currentElem + 10
+})
+console.log(increasedBy10)
+
+// 2. Given an array, return a new array where every non-string element is converted to an empty string.
+//
+// ```
+// Input: ["a", 123, "b", "c", {name: "cat"}]
+// Output: ["a", "", "b", "c", ""]
+// ```
+// // USE MAP METHOD
+let mixedArr = [1, 'a', 4, 'oops', {1: 'objectValue'}]
+let num2 = mixedArr.map((currentElem) => {
+  // console.log(typeof(currentElem))
+  if(typeof(currentElem) === 'string'){
+    return currentElem
+  }else{
+    return ''
+  }
+
+})
+console.log(num2)
+// 3. Write a function that returns only the even elements in an array of numbers.
+//
+// ```
+// Input: [1,2,3,4,5]
+// Output: [2,4]
+// ```
+// USE FILTER METHOD
+let evenArr = [3, 56, 35, 27, 4, 12]
+let itsEven = evenArr.filter((currentElem) =>{
+  return currentElem % 2 === 0
+})
+console.log(itsEven)
+
+// 4. Write a function that checks if all elements in the array are the same.
+//
+// ```
+// Input: [1,1,1,1,1]
+// Output: true
+// Input: [1,1,1,1,2]
+// Output: false
+// ```
+// USE EVERY ARRAY METHOD
+let sameArr = [2, 2, 2, 2, 2]
+let doubleChecked = sameArr.every((currentElem, i, arr) => {
+  if(currentElem === arr[0]){
+      return true
+    }else{
+      return false
+    }
+})
+console.log(doubleChecked)
+
+// 5. Write a function that returns the sum of every element in an array.
+//
+// ```
+// Input: [1,2,3,4,5]
+// Output: 15
+// ```
+// USE THE REDUCE METHOD
+let addArr = [2,7,24,27, 5]
+let added = addArr.reduce((currentElem, i) => {
+  return currentElem + i
+})
+console.log(added)
+
+// 6. Write a function that finds largest number in an array.
+//
+// ```
+// Input: [1,2,6,3,4,5]
+// Output: [6]
+// ```
+//USE THE REDUCE METHOD
+let makeItSmall = [1, 5, 7, 8, 19, 35, 2]
+let largest = makeItSmall.reduce((accumulator,crrElem)=>{
+  if(crrElem > accumulator){
+    return crrElem
+  } else{
+    return accumulator;
+  }
+})
+console.log(largest)
+
+// 7. Write a function `countZeroes`, which takes an array of numbers as its argument and returns the amount of zeroes that occur in it.
+//
+// ```
+// Input: [1,0,0,2,0,3,0,0,0]
+// Output: 6
+// ```
+
+const countZeroes = (arr) =>{
+  return arr.reduce((acc, currentElem)=>{
+    if(currentElem === 0){ // the accumulator is going to hold all of the zeroes
+      acc++
+      // console.log("ACC: " + acc)
+    }
+    return acc
+  }, 0) // setting the initial value of the acc to zero
+}
+
+// console.log("counting ZEROS: " + countZeroes([1,0,0,2,0,3,0,0,0]))
+
```

Comment:
I'm glad you have those notes on the side to keep you on track of what you want your function to exhibit .